src/realm/sync/noinst/client_impl_base.cpp

```diff
@@ -1889,10 +1891,10 @@ void Session::send_bind_message()
     REALM_ASSERT_EX(m_state == Active, m_state);
 
     session_ident_type session_ident = m_ident;
-    bool need_client_file_ident = !have_client_file_ident();
+    // Request an ident if we don't already have one and there isn't a pending client reset diff
+    bool need_client_file_ident = !have_client_file_ident() && !get_client_reset_config();
```

Comment:
don't think the second check is needed since you now have a file ident when get_client_reset_config() is true

Reply:
It is needed, since we don't want to send the IDENT message right away and there are some tests in `test_client_reset.cpp` that do not set an initial file ident for the local realm when the client reset diff is performed by passing in a client reset config struct.

Reply:
got it. the tests seem wrong though. I don't think there can be a client reset when the app connects to the server for the first time (and request a file ident).

Reply:
There was 1 sync test that started with a 0 file ident when performing a client reset and that was updated.
However, for the PBS->FLX migration tests, there is a case where we can get a client reset when the file ident is 0: if we start with a brand new realm or are using Async open and the migration is triggered upon first connect to the server.

I kept the logic as is and added a comment about migrations.